engine-tests/src/test/java/org/terasology/engine/integrationenvironment/MTEExtensionTestWithPerMethodLifecycle.java

```diff
@@ -23,30 +23,26 @@
 import java.util.List;
 import java.util.UUID;
 import java.util.concurrent.ConcurrentHashMap;
-import java.util.concurrent.ConcurrentMap;
 
+import static com.google.common.truth.Truth.assertThat;
 import static org.junit.jupiter.api.Assertions.assertEquals;
-import static org.junit.jupiter.api.Assertions.assertFalse;
-import static org.junit.jupiter.api.Assertions.assertNotNull;
-import static org.junit.jupiter.api.Assertions.assertTrue;
 
 /**
- * Ensure a test class with a per-method Jupiter lifecycle can share an engine between tests.
+ * Ensure a test class with a per-method Jupiter lifecycle does not share data between tests.
  */
 @Tag("MteTest")
 @ExtendWith(MTEExtension.class)
 @TestInstance(TestInstance.Lifecycle.PER_METHOD)  // The default, but here for explicitness.
 @TestMethodOrder(MethodOrderer.OrderAnnotation.class)
 public class MTEExtensionTestWithPerMethodLifecycle {
-    // java 8 doesn't have ConcurrentSet
-    @SuppressWarnings("checkstyle:constantname")
-    private static final ConcurrentMap<String, Integer> seenNames = new ConcurrentHashMap<>();
+    @SuppressWarnings("checkstyle:ConstantName")
+    private static final ConcurrentHashMap.KeySetView<Object, Boolean> seenNames = ConcurrentHashMap.newKeySet();
```

Comment:
If non-static, this member will get re-initialized with each lifecycle, i.e., `PER_METHOD`. When static, it seems to be carried over, so the `thingsDoNotPolluteSecondTest` fails as the list still contains the name from the first test.

```suggestion
    private final ConcurrentHashMap.KeySetView<Object, Boolean> seenNames = ConcurrentHashMap.newKeySet();
```